xbmc/dialogs/GUIDialogSelectColor.cpp

```diff
@@ -0,0 +1,239 @@
+        Close();
+      }
+    }
+    break;
+
+    case GUI_MSG_SETFOCUS:
+    {
+      if (m_viewControl.HasControl(message.GetControlId()))
+      {
+        if (m_vecList->IsEmpty())
+        {
+          SET_CONTROL_FOCUS(CONTROL_CANCEL_BUTTON, 0);
+          return true;
+        }
+        if (m_viewControl.GetCurrentControl() != message.GetControlId())
+        {
+          m_viewControl.SetFocused();
+          return true;
+        }
+      }
+    }
+    break;
+  }
+
+  return CGUIDialogBoxBase::OnMessage(message);
+}
+
+void CGUIDialogSelectColor::OnSelect(int idx)
+{
+  m_bConfirmed = true;
+  Close();
+}
+
+bool CGUIDialogSelectColor::OnBack(int actionID)
+{
+  m_vecList->Clear();
+  m_bConfirmed = false;
+  return CGUIDialogBoxBase::OnBack(actionID);
+}
+
+void CGUIDialogSelectColor::Reset()
+{
+  m_focusToButton = false;
+  m_selectedColor.clear();
+  m_vecList->Clear();
+}
+
+std::string CGUIDialogSelectColor::GetSelectedColor() const
+{
+  return m_selectedColor;
+}
+
+int CGUIDialogSelectColor::GetSelectedItem() const
+{
+  for (int index = 0; index < m_vecList->Size(); index++)
+  {
+    if (StringUtils::CompareNoCase(m_selectedColor, m_vecList->Get(index)->GetLabel2()) == 0)
+    {
+      return index;
+    }
+  }
+  return -1;
+}
+
+void CGUIDialogSelectColor::SetSelectedColor(const std::string& colorHex)
+{
+  if (colorHex.empty())
+    m_selectedColor = "FF000000";
+  else
+    m_selectedColor = colorHex;
+}
+
+void CGUIDialogSelectColor::SetButtonFocus(bool buttonFocus)
+{
+  m_focusToButton = buttonFocus;
+}
+
+CGUIControl* CGUIDialogSelectColor::GetFirstFocusableControl(int id)
+{
+  if (m_viewControl.HasControl(id))
+    id = m_viewControl.GetCurrentControl();
+  return CGUIDialogBoxBase::GetFirstFocusableControl(id);
+}
+
+void CGUIDialogSelectColor::OnWindowLoaded()
+{
+  CGUIDialogBoxBase::OnWindowLoaded();
+  m_viewControl.Reset();
+  m_viewControl.SetParentWindow(GetID());
+  m_viewControl.AddView(GetControl(CONTROL_ICON_LIST));
+}
+
+void CGUIDialogSelectColor::OnInitWindow()
+{
+  // If no file path provided, load the default file
+  if (m_filePath.empty())
+    m_filePath =
+        CSpecialProtocol::TranslatePathConvertCase("special://xbmc/system/dialogcolors.xml");
+
+  std::vector<std::pair<std::string, UTILS::ColorEx>> colors;
+  if (UTILS::COLORUTILS::LoadXMLFileColors(m_filePath, colors, true))
+  {
+    for (auto& color : colors)
+    {
+      CFileItem* item = new CFileItem(color.first);
+      item->SetLabel2(color.second.hex);
+      m_vecList->Add(CFileItemPtr(item));
+    }
+  }
+
+  m_viewControl.SetItems(*m_vecList);
+  m_viewControl.SetCurrentView(CONTROL_ICON_LIST);
+  
+  SET_CONTROL_LABEL(CONTROL_NUMBER_OF_ITEMS,
+                    StringUtils::Format("{} {}", m_vecList->Size(), g_localizeStrings.Get(127)));
+
+  SET_CONTROL_LABEL(CONTROL_CANCEL_BUTTON, g_localizeStrings.Get(222));
+
+  CGUIDialogBoxBase::OnInitWindow();
+
+  // focus one of the buttons if explicitly requested
```

Comment:
Then this block of code should be guarded against something that can only be set if the codeflow has hit window initialize (maybe a null check in the color vector is enough). Log an error if it doesnt